### ISSUE TITLE
Improve handling of WordPress post embeds

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#809]
The handling of post embeds isn't great in AMP right now because the `p.wp-embedded-content` isn't getting replaced with the `iframe.wp-embedded-content` like it does in the non-AMP version. The result is some doubling of content.

Compare in a post I just published:

[Canonical (non-AMP) version](https://weston.ruter.net/2017/11/16/mel-choyce-recap-on-wordpress-4-9/):

> ![image](https://user-images.githubusercontent.com/134745/32963820-0accfe62-cb86-11e7-9f63-0f3272a7cbb8.png)

[AMP version](https://weston.ruter.net/2017/11/16/mel-choyce-recap-on-wordpress-4-9/amp/):

> ![image](https://user-images.githubusercontent.com/134745/32963799-fa7c7b82-cb85-11e7-9231-c988b2e227c3.png)

**Comment by westonruter** [comment]
Things have regressed in the current version of the plugin. The iframe is not showing up at all now.

Given `post_content` that looks like:

```
See post on Make XWP:

https://make.xwp.co/2018/09/24/amp-plugin-release-v1-0-beta4/
```

The rendered AMP response looks like this:

![image](https://user-images.githubusercontent.com/134745/45983184-8220d680-c010-11e8-8b33-44660239d712.png)

There is actually still an `iframe` there:

```html
<amp-iframe
class="wp-embedded-content amp-wp-enforced-sizes amp-wp-5e9b0a2"
sandbox="allow-scripts"
src="https://make.xwp.co/2018/09/24/amp-plugin-release-v1-0-beta4/embed/#?secret=swkquRFsjz"
data-secret="swkquRFsjz"
width="600" 
height="338"
title="“AMP Plugin Release v1.0-beta4” — Make XWP" 
frameborder="0" 
scrolling="no" 
layout="intrinsic">
<div placeholder="" class="amp-wp-iframe-placeholder"></div></amp-iframe>
```

But it is now showing up because of this CSS:

```css
.amp-wp-5e9b0a2{position:absolute;clip:rect(1px,1px,1px,1px)}
```

This is coming from core: https://github.com/WordPress/wordpress-develop/blob/45efe978fc7ed9ae865b1a80a16aef6570fca0e1/src/wp-includes/embed.php#L783

**Comment by westonruter** [comment]
Proposal to implement AMP component for embedding WordPress posts: https://github.com/ampproject/amphtml/issues/18378

**Comment by jerclarke** [comment]
There's a lot of tickets about WP embeds in AMP, but it seems like this one is the "parent" that's still open. 

Just want to confirm current status as a user: **This work is still ongoing, and for now WP embeds in AMP simply don't work, and will show as the basic blockquote.**

If so that's fine. Luckily the WP embed has a great fallback format (unlike FB and most of the others). Looking forward to the day when all this work you're doing pays off :)

**Comment by westonruter** [comment]
@jerclarke That's correct. The ultimate resolution for this issue is to introduce a new `amp-wordpress-embed` component in AMP itself: https://github.com/ampproject/amphtml/pull/24952

Once that is available, we can finalize https://github.com/ampproject/amp-wp/pull/3465 which will close this issue.

I don't have a good timetable on when we'll get to this. Perhaps April?

**Comment by jerclarke** [comment]
Awesome. Safe pitch to my org to keep using the embeds and be patient for an AMP-native solution in the near future 🙏🏻

**Comment by westonruter** [comment]
This is no longer blocked! The `amp-wordpress-embed` component is now a thing as of https://github.com/ampproject/amphtml/pull/34948. 

See [example in AMP Playground](https://playground.amp.dev/#share=PCFET0NUWVBFIGh0bWw+CjxodG1sIOKaoT4KPGhlYWQ+CiAgPG1ldGEgY2hhcnNldD0idXRmLTgiIC8+CiAgPHRpdGxlPk15IEFNUCBQYWdlPC90aXRsZT4KICA8bGluayByZWw9ImNhbm9uaWNhbCIgaHJlZj0ic2VsZi5odG1sIiAvPgogIDxtZXRhIG5hbWU9InZpZXdwb3J0IiBjb250ZW50PSJ3aWR0aD1kZXZpY2Utd2lkdGgiIC8+CiAgPHN0eWxlIGFtcC1ib2lsZXJwbGF0ZT5ib2R5ey13ZWJraXQtYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1vei1hbmltYXRpb246LWFtcC1zdGFydCA4cyBzdGVwcygxLGVuZCkgMHMgMSBub3JtYWwgYm90aDstbXMtYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7YW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGh9QC13ZWJraXQta2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUAtbW96LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1zLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW8ta2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUBrZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19PC9zdHlsZT48bm9zY3JpcHQ+PHN0eWxlIGFtcC1ib2lsZXJwbGF0ZT5ib2R5ey13ZWJraXQtYW5pbWF0aW9uOm5vbmU7LW1vei1hbmltYXRpb246bm9uZTstbXMtYW5pbWF0aW9uOm5vbmU7YW5pbWF0aW9uOm5vbmV9PC9zdHlsZT48L25vc2NyaXB0PgogIDxzY3JpcHQgYXN5bmMgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC5qcyI+PC9zY3JpcHQ+CiAgPHNjcmlwdCBhc3luYyBjdXN0b20tZWxlbWVudD0iYW1wLXdvcmRwcmVzcy1lbWJlZCIgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC9hbXAtd29yZHByZXNzLWVtYmVkLTEuMC5qcyI+PC9zY3JpcHQ+CjwvaGVhZD4KPGJvZHk+CiAgPGFtcC13b3JkcHJlc3MtZW1iZWQgZGF0YS11cmw9Imh0dHBzOi8vbWFrZS53b3JkcHJlc3Mub3JnL2NvcmUvMjAxNS8xMC8yOC9uZXctZW1iZWRzLWZlYXR1cmUtaW4td29yZHByZXNzLTQtNC8iIGxheW91dD0iZml4ZWQtaGVpZ2h0IiBoZWlnaHQ9IjQwMCI+CiAgICA8YnV0dG9uIG92ZXJmbG93PkxvYWQgbW9yZTwvYnV0dG9uPgogIDwvYW1wLXdvcmRwcmVzcy1lbWJlZD4KPC9ib2R5Pgo8L2h0bWw+).

Work can proceed once https://github.com/ampproject/amp-wp/pull/6436 is merged or by basing off of that branch.

**Comment by westonruter** [comment]
> Work can proceed once #6436 is merged or by basing off of that branch.

The PR has been merged into `develop`. This work can proceed.

**Comment by westonruter** [comment]
Punting since the bento experiment is still required: https://tundra-functional-double.glitch.me/

![image](https://user-images.githubusercontent.com/134745/144467430-507fb87c-0ccf-4491-95b2-54ccef334f27.png)

**Comment by pooja-muchandikar** [comment]
QA Passed ✅ The Post Embeds looks fine. 

Before: Posts embeds were not rendering properly

<img width="1076" alt="Screenshot 2022-03-22 at 1 08 07 PM" src="https://user-images.githubusercontent.com/41000648/159430969-1204b475-9e33-488e-be54-8014ecf0f0f0.png">

---

After: Posts embeds are rendering properly

<img width="1046" alt="Screenshot 2022-03-22 at 1 07 58 PM" src="https://user-images.githubusercontent.com/41000648/159431046-96c78cd5-5347-4f5d-9b4e-e7bf3ed640c4.png">

---

But one observation here:

As per [comment](https://github.com/ampproject/amp-wp/pull/6665#issuecomment-954944600) the expand button persist or instead the element should increase it's height as soon as it receives the message from the origin. But seems like it's expected behavior.

**Comment by westonruter** [comment]
> As per [comment](https://github.com/ampproject/amp-wp/pull/6665#issuecomment-954944600) the expand button persist or instead the element should increase it's height as soon as it receives the message from the origin. But seems like it's expected behavior.

If the embed is in the (initial) viewport when it loads, then it's expected that it would not resize since this would cause a layout shift.
